src/code_file_manager.rb

```diff
@@ -1,8 +1,9 @@
+require 'tempfile'
+
 module CodeFileManager
-  def next_file_absolute_path
-    possible_letters = [('a'..'z'), ('A'..'Z')].map(&:to_a).flatten
-    random_string = (0..10).map { possible_letters[rand(possible_letters.size)] }.join
-    "#{Dir.pwd}/codes/#{random_string}"
+  def next_file_absolute_path(extension)
+    tempfile = Tempfile.create(['foo', extension], "#{Dir.pwd}/codes")
```

Comment:
I don't really like the way you're using Tempfile. IMHO, the class CodeFileManager is doing pretty much the same as Tempfile, and it's not really abstracting anything. 

AFAIK, already creates and returns a file, and you're actually creating it just to get the path (and open the file again later). This is problematic because you might not be able to open the file again (e.g if the name was used by another process between the time you generate the file name and the time you open the file, or you might not be able to open the file in write mode because it was already opened by your own process). The magic of this kind of libraries is that they give you a file object, so you don't have to deal with these race conditions, and that's the way you should use it.

IMHO, you should remove the CodeFileManager class, and use Tempfile directly when you write the code to a file, just as indicated in the example of Tempfile: https://ruby-doc.org/stdlib-2.4.1/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close